sources/packages/api/src/services/bceid/bceid.service.ts

```diff
@@ -101,6 +93,77 @@ export class BCeIDService {
     }
   }
 
+  /**
+   * Retrieves business BCeIDs accounts under the specified institution (options.businessGuid)
+   * and that the user (options.requesterUserGuid) executing the search has access to.
+   * @param options Search parameters.
+   * @returns The list of BCeIDs for the institution.
+   */
+  public async searchBCeIDAccounts(
+    options: SearchAccountOptions,
+  ): Promise<SearchBCeIDAccountResult> {
+    var client = await this.getSoapClient();
+    // SOAP call body to execute the searchBCeIDAccount request.
+    var body = {
+      bceidAccountSearchRequest: {
+        onlineServiceId: this.bceidConfig.onlineServiceId,
+        requesterAccountTypeCode: BCeIDAccountTypeCodes.Business,
+        requesterUserGuid: options.requesterUserGuid,
+        pagination: {
+          pageSizeMaximum: options.pagination.pageSize,
+          pageIndex: options.pagination.pageIndex,
+        },
+        sort: {
+          direction: SortDirection.Ascending,
+          onProperty: SortBCeIDAccountOnProperty.Firstname,
+        },
+        accountMatch: {
+          searchableAccountType: BCeIDAccountTypeCodes.Business,
+        },
+        businessMatch: {
+          businessGuid: options.businessGuid,
+        },
+      },
+    };
+
+    try {
+      // Destructuring the result of the SOAP request to get the
+      // first item on the array where the parsed js object is.
+      const [result] = await client.searchBCeIDAccountAsync(body);
+      const response = result.searchBCeIDAccountResult as ResponseBase;
+      this.ensureSuccessStatusResult(response);
+      // Array os items from SOAP response that contains all accounts.
+      const accounts = result.searchBCeIDAccountResult.accountList.BCeIDAccount;
+      const pagination = result.searchBCeIDAccountResult.pagination;
+
+      const accountResults = accounts.map((account: any) => {
```

Comment:
Can we create any type regarding account?

Reply:
I did not add before because the SOAP structure that is mapped is a little complex in terms of nesting objects but since it could bring more clarity to the code, I created 5 new types to map this. Thanks for the recommendation 👍